### ISSUE TITLE
set cookie in middleware

### ISSUE THREAD

**Issue by EduardoMateos** [andreizanik/cookies-next#12]
Hello,

is it possible to set a cookie in a middleware?
I do not give with it!

thanks for your job!

**Comment by NetSpud** [comment]
I have the same problem - attempting to set a cookie in a middleware file just won't work. Setting in API routes is absolutely fine.
To recreate, try setting setting a cookie in a middleware file

### package.json
```"dependencies": {
    "@prisma/client": "^3.10.0",
    "@shopify/react-csrf": "^2.1.9",
    "bcrypt": "^5.0.1",
    "cookies-next": "^2.0.4",
    "next": "12.1.0",
    "next-csrf": "^0.1.2",
    "node-cron": "^3.0.0",
    "prisma": "^3.10.0",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  },
  "devDependencies": {
    "@types/bcrypt": "^5.0.0",
    "@types/node": "17.0.21",
    "@types/react": "17.0.39",
    "autoprefixer": "^10.4.2",
    "eslint": "8.9.0",
    "eslint-config-next": "12.1.0",
    "postcss": "^8.4.7",
    "tailwindcss": "^3.0.23",
    "typescript": "^4.5.5"
  }
```

**Comment by andreizanik** [comment]
Hey! I didn't even take the time to test setting cookies in the middleware.

I will try to solve this problem in the near future

**Comment by santyas** [comment]
> Hello,
> 
> is it possible to set a cookie in a middleware? I do not give with it!
> 
> thanks for your job!

Meanwhile try to use the NextRequest middleware param. For example, destructuring that request:
`const {cookies} = request`

Then:
`console.log(`Locale cookie: ${cookies['NEXT_LOCALE']}`)`

And to Set:
response.cookie('NEXT_LOCALE', 'ar')
then:
return response

**Comment by 7iomka** [comment]
Any news with this?
I want to use your plugin as a replacement for nookies

**Comment by andreizanik** [comment]
@7iomka 
Sorry, I don't have enough time to complete this task.
But nextjs documentation has an example of how you can work with cookies
https://nextjs.org/docs/advanced-features/middleware#using-cookies

**Comment by sohaibMan** [comment]
> Hey! I didn't even take the time to test setting cookies in the middleware.
> 
> I will try to solve this problem in the near future

did you complete this ? I want to work for this feature

**Comment by andreizanik** [comment]
> did you complete this ? I want to work for this feature 
No, I didn't
If you want, make a PR. I'll be glad

**Comment by greg2012201** [comment]
I have been recently considering how to implement support for Next.js 13 middleware. If I find some spare time, I will work on it and submit a pull request with the implementation.

**Comment by andreizanik** [comment]
Thanks @greg2012201 for adding support in middleware. 
I published version 4.0.0
